python/ray/data/dataset.py

```diff
@@ -2458,6 +2459,16 @@ def write_datasource(
         """
 
         ctx = DatasetContext.get_current()
+        if ray_remote_args is None:
+            ray_remote_args = {}
+        if not ray.util.client.ray.is_connected():
```

Comment:
Should we raise an error if a local scheme is provided but the user is using Ray Client?

Reply:
It was treated as non-custom path in Ray Client. But sounds good to raise error.